src/plugins/meowcrypt/index.tsx

```diff
@@ -0,0 +1,346 @@
+                            >
+                                <path d="M 6.75 6.75 L 6.75 9 L 14.25 9 L 14.25 6.75 C 14.25 4.679688 12.570312 3 10.5 3 C 8.429688 3 6.75 4.679688 6.75 6.75 Z M 3.75 9 L 3.75 6.75 C 3.75 3.023438 6.773438 0 10.5 0 C 14.226562 0 17.25 3.023438 17.25 6.75 L 17.25 9 L 18 9 C 19.65625 9 21 10.34375 21 12 L 21 21 C 21 22.65625 19.65625 24 18 24 L 3 24 C 1.34375 24 0 22.65625 0 21 L 0 12 C 0 10.34375 1.34375 9 3 9 Z M 3.75 9" />
+                                {!enabled && <line x1="22" y1="2" x2="2" y2="22" stroke="var(--red-500)" stroke-width="2.5" />}
+                            </svg>
+                        </div>
+                    </Button>
+                </div>
+            )}
+        </Tooltip>
+    );
+}
+
+// thanks GreetStickerPicker plugin :3
+function ProfilesMenu() {
+    const [signal, refetchProfile] = React.useReducer(x => x + 1, 0);
+    const [currentProfile] = useAwaiter(getCurrentProfile, { deps: [signal], fallbackValue: "Public" });
+
+    const [profiles] = useAwaiter(async () => {
+        const p = await getProfiles();
+        let matchedPublic = false;
+        p.forEach(async profile => {
+            if (profile.name === "Public") matchedPublic = true;
+        });
+        if (!matchedPublic) {
+            const p2 = await addProfile(publicProfile);
+            return p2;
+        }
+        return p;
+    }, { fallbackValue: [publicProfile] });
+
+    return (
+        <Menu.Menu
+            navId="meowcrypt-profiles"
+            onClose={() => FluxDispatcher.dispatch({ type: "CONTEXT_MENU_CLOSE" })}
+            aria-label="Meowcrypt Profiles"
+        >
+            <Menu.MenuGroup
+                label="Meowcrypt"
+            >
+                <Menu.MenuItem id="profiles" label="Profiles">
+                    {profiles.map(profile => (
+                        <Menu.MenuRadioItem
+                            key={profile.name}
+                            group="meowcrypt-profile"
+                            id={"meowcrypt-profile-" + profile.name}
+                            label={profile.name}
+                            checked={profile.name === currentProfile}
+                            action={async () => await setCurrentProfile(profile.name).then(refetchProfile)}
+                        />
+                    ))}
+                </Menu.MenuItem>
+
+                <Menu.MenuItem
+                    key="add-profile"
+                    id="meowcrypt-add-profile"
+                    label="Add Profile"
+                    action={() => buildAddProfileModal()}
+                />
+
+                <Menu.MenuItem
+                    key="delete-profile"
+                    id="meowcrypt-delete-profile"
+                    label="Delete Current Profile"
+                    action={async () => await removeProfile(currentProfile).then(async () => await setCurrentProfile("Public").then(refetchProfile))}
```

Comment:
there is no need for these awaits
```suggestion
                    action={() => 
                    	removeProfile(currentProfile)
                    		.then(() => setCurrentProfile("Public"))
                    		.then(refetchProfile)
              		}
```